### PR TITLE
Add sound support for single media export

### LINKED ISSUE #1308: Convert sounds with singlefile converter
**Required skills**: Python, maybe Cython

**Difficulty**: Easy

Our exporter for single media files can only be used for graphics files (SLP, SMP or SMX) currently. It would be great, if we could extend it to support outputting sounds as well. This functionality can reuse the internal opus conversion service that is already implemented.

**Further reading**:

* [Singlefile converter code](https://github.com/SFTtech/openage/blob/master/openage/convert/tool/singlefile.py)
* [Converter architecture](https://github.com/SFTtech/openage/blob/master/doc/code/converter/architecture_overview.md)
* [Converter workflow](https://github.com/SFTtech/openage/blob/master/doc/code/converter/workflow.md)

### PR DESCRIPTION
**Pull request by drs-11**
Fixes #1308.

### CODE REVIEW

**Comment by heinezen** [comment]
@drs-11 Already done? :)

**Comment by drs-11** [comment]
Yeah I thought this was a simple one. I didn't underestimate this did I?

**Comment by heinezen** [comment]
@drs-11 Just wanted to make sure that you are done before I review (because you still have the `WIP` tag in the title)

**Comment by drs-11** [comment]
@heinezen the things I have done so far: 
- added wav to opus methods (single file and from drs archive) 
- modified main function to include the added methods
- changed comments and info to describe the sound export methods

If that's all that was needed to be done then please go ahead and review otherwise please mention things I missed.

File: openage/convert/tool/singlefile.py [hunk 1]
```diff
@@ -42,11 +42,11 @@ def main(args, error):
     file_path = Path(args.filename)
     file_extension = file_path.suffix[1:].lower()
 
-    if not args.palettes_path:
-        raise Exception("palettes-path needs to be specified")
-
-    palettes_path = Path(args.palettes_path)
-    palettes = read_palettes(palettes_path)
+    if args.mode not in ["drs-wav", "wav"]:
```
Comment (heinezen) [inline openage/convert/tool/singlefile.py:45]: This also needs to check the file extension.

File: openage/convert/tool/singlefile.py [hunk 2]
```diff
@@ -42,11 +42,11 @@ def main(args, error):
...
+        if not args.palettes_path:
+            raise Exception("palettes-path needs to be specified")
+        palettes_path = Path(args.palettes_path)
```
Comment (heinezen) [inline openage/convert/tool/singlefile.py:48]: Newline in between these lines, please

**Review by heinezen — CHANGES_REQUESTED** [review]
Found one small problem and a formatting error, but looks good otherwise :)

@drs-11 Btw, if you think you are done, then you should request a review in the Github UI (top right) or remove the `[WIP]` label. Then we know that you are ready.

File: openage/convert/tool/singlefile.py [hunk 3]
```diff
@@ -42,11 +42,12 @@ def main(args, error):
     file_path = Path(args.filename)
     file_extension = file_path.suffix[1:].lower()
 
-    if not args.palettes_path:
-        raise Exception("palettes-path needs to be specified")
+    if args.mode not in ["drs-wav", "wav"] or file_extension != "wav":
```
Comment (heinezen) [inline openage/convert/tool/singlefile.py:45]: This condition will still be `True` if no `args.mode` was given, but the file extension is `wav`.

Better use `not (args.mode in ["drs-wav", "wav"] or file_extension == "wav")`

**Review by heinezen — CHANGES_REQUESTED** [review]

File: openage/convert/tool/singlefile.py (hunk 3 above)
Comment (drs-11) [reply openage/convert/tool/singlefile.py:45]: ok yeah

**Review by heinezen — APPROVED** [review]
Now it works like a charm!

File: openage/convert/tool/singlefile.py (hunk 3 above)
Comment (heinezen) [reply openage/convert/tool/singlefile.py]: Boolean logic :D

**Pull request merged**